src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -1629,7 +1629,7 @@ bool ExecutorWorker::hasRight(SyncOpPtr syncOp, bool &exists) {
                     return false;
                 }
 
-                return _syncPal->_remoteSnapshot->canWrite(*newCorrespondingParentNode->id());
+                return true;
```

Comment:
The methods impacted by your changes are essentially consuming a `SyncOp` object. Therefore they should be easily unit-testable. However, because of the logging operations and macros, the methods require more data to be actually tested. A small refactoring could help. Do you think it's manageable given our timeline?

Reply:
Done in 6a06e7fa011a0bb5cf5f7e4c76b32f03e8583b6a